File: iPad_Console_2_0_0_Betas_BeforeButtonEnhancements/Classes/GridLayoutContainerView.m
/* OpenRemote, the Home of the Digital Home.
 *  * Copyright 2008-2011, OpenRemote Inc-2009, OpenRemote Inc.
 * 
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */
#import "GridLayoutContainerView.h"
#import "GridLayoutContainer.h"
#import "GridCellView.h"
#import "GridCell.h"

@implementation GridLayoutContainerView

// Override method of UIView and be called automatically.
- (void)layoutSubviews
{
	GridLayoutContainer *grid = (GridLayoutContainer *)layout;
	int h = self.bounds.size.height / grid.rows;				
	int w = self.bounds.size.width / grid.cols;
	
	if (grid) {
		for (GridCell *cell in grid.cells){
			GridCellView *cellView = [[GridCellView alloc] initWithGridCell:cell frame:CGRectMake(cell.x * w, cell.y * h, w * cell.colspan, h * cell.rowspan)];
			[self addSubview:cellView];
            [cellView release];
		}
	}
}

@end

File: branches/project/WebConsole_1_0_0_Alphas/src/org/openremote/web/console/client/view/GridLayoutContainerView.java
/* OpenRemote, the Home of the Digital Home.
* Copyright 2008-2010, OpenRemote Inc.
*
* See the contributors.txt file in the distribution for a
* full listing of individual contributors.
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU Affero General Public License as
* published by the Free Software Foundation, either version 3 of the
* License, or (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
* GNU Affero General Public License for more details.
*
* You should have received a copy of the GNU Affero General Public License
* along with this program. If not, see <http://www.gnu.org/licenses/>.
*/
package org.openremote.web.console.client.view;

import org.openremote.web.console.domain.GridCell;
import org.openremote.web.console.domain.GridLayoutContainer;

import com.extjs.gxt.ui.client.widget.LayoutContainer;

/**
 * Wraps grid cells, and display in a absolute position.
 */
public class GridLayoutContainerView extends LayoutContainer {

   public GridLayoutContainerView(GridLayoutContainer gridLayoutContainer) {
      setStyleAttribute("position", "absolute");
      setSize(gridLayoutContainer.getWidth(), gridLayoutContainer.getHeight());
      setPosition(gridLayoutContainer.getLeft(), gridLayoutContainer.getTop());
      init(gridLayoutContainer);
   }
   
   private void init(GridLayoutContainer gridLayoutContainer) {
      int cellWidth = gridLayoutContainer.getWidth() / gridLayoutContainer.getCols();
      int cellHeight = gridLayoutContainer.getHeight() / gridLayoutContainer.getRows();
      for (GridCell gridCell: gridLayoutContainer.getCells()) {
         add(new GridCellContainerView(cellWidth, cellHeight, gridCell));
      }
   }
}
